Cannot link against static openssl libs on Windows (gdi32 linkage missing)

Linking with static openssl fails on Windows exactly like in #293. E.g.:
```
  = note: libopenssl_sys-832ea8393e41d5b9.rlib(rand_win.obj) : error LNK2019: unresolved external symbol __imp_CreateCompatibleBitmap referenced in function RAND_screen
          libopenssl_sys-832ea8393e41d5b9.rlib(rand_win.obj) : error LNK2019: unresolved external symbol __imp_DeleteObject referenced in function RAND_screen
          libopenssl_sys-832ea8393e41d5b9.rlib(rand_win.obj) : error LNK2019: unresolved external symbol __imp_GetDeviceCaps referenced in function RAND_screen
          libopenssl_sys-832ea8393e41d5b9.rlib(rand_win.obj) : error LNK2019: unresolved external symbol __imp_GetDIBits referenced in function RAND_screen
          libopenssl_sys-832ea8393e41d5b9.rlib(rand_win.obj) : error LNK2019: unresolved external symbol __imp_GetObjectW referenced in function RAND_screen
          C:\path\to\project\target\debug\deps\project-fcf57d0de9731d4b.exe : fatal error LNK1120: 5 unresolved externals
```

https://github.com/sfackler/rust-openssl/commit/9ba96b80b171d3d869b33529867668b36533b05f removed the import of `gdi32-sys`, reverting PR https://github.com/sfackler/rust-openssl/pull/297.

If I add `gdi32-sys` as a dependency _to my own project_ everything links successfully again.
Can we and should we thus revert that commit?
Would it correctly fix the problem generally in this project even without any `extern crate` as it was before? (@retep998?)

----

Steps to reproduce:
* Install `vcpkg`
* Run `vcpkg install openssl:x64-windows-static`
* Set `OPENSSL_STATIC` and point `OPENSSL_DIR` to the `$vcpkg\installed\x64-windows-static\lib` directory
* Try to build a project depending on rust-openssl

--- a/openssl-sys/build/main.rs
+++ b/openssl-sys/build/main.rs
@@ -100,6 +100,10 @@ fn main() {
     for lib in libs.into_iter() {
         println!("cargo:rustc-link-lib={}={}", kind, lib);
     }
+
+    if kind == "static" && target.contains("windows") {
+        println!("cargo:rustc-link-lib=dylib=gdi32");
+    }
 }
 
 fn find_openssl_dir(target: &str) -> OsString {

